### PR TITLE
r/codebuild_report_group - add new resource

### LINKED ISSUE #12178: Test Reports with AWS CodeBuild
### Community Note
* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description
AWS extended the Codebuild Service with new functionality: Test Reports
The Report Groups can be populated form within a build by exporting JUnit or Cucumber test reports. Also, these reports can be persisted to S3 if specified on the Report Group.

_"AWS CodeBuild announced the launch of a new feature in CodeBuild called Reports. This feature allows you to view the reports generated by functional or integration tests. The reports can be in the JUnit XML or Cucumber JSON format. You can view metrics such as Pass Rate %, Test Run Duration, and number of Passed versus Failed/Error test cases in one location. Builders can use any testing frameworks as long as the reports are generated in the supported formats.
You can see the test reports created by CodeBuild in the respective Report Group, where they are stored for 30 days. For longer retention, you can store the reports in an Amazon S3 bucket. Each test report is further broken down by individual test cases."_

Also, boto3 already supports the new functionality.

### New or Affected Resource(s)
* aws_codebuild_reportgroup

### Potential Terraform Configuration
```hcl
resource aws_codebuild_reportgroup sample {
 name = "my-report-group"
 type = "TEST"
 exportConfig = {
  exportConfigType: "S3" | "NO_EXPORT"
  s3Destination: {
   bucket: "my-bucket"
   path: "my-custom-prefix"
   packaging: "ZIP" | "NONE"
   encryptionKey: "my-kms-key"
   encryptionDisabled: True | False
   }
  }
}
```

### References
https://aws.amazon.com/blogs/devops/test-reports-with-aws-codebuild/
https://boto3.amazonaws.com/v1/documentation/api/latest/reference/services/codebuild.html

### PR DESCRIPTION
**Pull request by DrFaust92**
### Community Note

* Please vote on this pull request by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original pull request comment to help the community and maintainers prioritize this request
* Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for pull request followers and do not help prioritize the request

Closes #12178

Release note for [CHANGELOG](https://github.com/terraform-providers/terraform-provider-aws/blob/master/CHANGELOG.md):

```release-note
resource_aws_codebuild_report_group - add new resource
```

Output from acceptance testing:

```
$ make testacc TESTARGS='-run=TestAccAWSCodeBuildReportGroup_'
--- PASS: TestAccAWSCodeBuildReportGroup_export_s3 (121.99s)
--- PASS: TestAccAWSCodeBuildReportGroup_tags (89.06s)
--- PASS: TestAccAWSCodeBuildReportGroup_disappears (28.05s)
--- PASS: TestAccAWSCodeBuildReportGroup_basic (37.50s)
```

### CODE REVIEW

**Comment by ewbankkit** [comment]
Verified acceptance tests:

```console
$ make testacc TEST=./aws TESTARGS='-run=TestAccAWSCodeBuildReportGroup_'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./aws -v -count 1 -parallel 20 -run=TestAccAWSCodeBuildReportGroup_ -timeout 120m
=== RUN   TestAccAWSCodeBuildReportGroup_basic
=== PAUSE TestAccAWSCodeBuildReportGroup_basic
=== RUN   TestAccAWSCodeBuildReportGroup_updated
=== PAUSE TestAccAWSCodeBuildReportGroup_updated
=== RUN   TestAccAWSCodeBuildReportGroup_disappears
=== PAUSE TestAccAWSCodeBuildReportGroup_disappears
=== CONT  TestAccAWSCodeBuildReportGroup_basic
=== CONT  TestAccAWSCodeBuildReportGroup_disappears
=== CONT  TestAccAWSCodeBuildReportGroup_updated
--- PASS: TestAccAWSCodeBuildReportGroup_disappears (61.10s)
--- PASS: TestAccAWSCodeBuildReportGroup_basic (65.39s)
--- PASS: TestAccAWSCodeBuildReportGroup_updated (92.03s)
PASS
ok  	github.com/terraform-providers/terraform-provider-aws/aws	92.073s
```

**Comment by rubytobi** [comment]
@DrFaust92 awesome, thx for implementing this! 
Just saw that AWS rolled out (tagging for report groups) [https://awsapichanges.info/archive/changes/039f9b-codebuild.html]. Do you think you could add support for this as well?
Maybe it's worth pinging some reviewers afterwards - almost two months since your implementation :)

**Comment by DrFaust92** [comment]
Hey @rubytobi, i can address this when https://github.com/terraform-providers/terraform-provider-aws/pull/13459 is merged.

File: aws/resource_aws_codebuild_report_group.go [hunk 1]
```diff
@@ -0,0 +1,283 @@
...
+}
+
+func resourceAwsCodeBuildReportGroupUpdate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).codebuildconn
+
+	input := &codebuild.UpdateReportGroupInput{
+		Arn: aws.String(d.Id()),
+	}
+
+	updateNeeded := false
```
Comment (ewbankkit) [inline aws/resource_aws_codebuild_report_group.go:168]: Is this necessary? `resourceAwsCodeBuildReportGroupUpdate()` won't be called unless at least one of `export_config` or `tags` has changed.

Comment (DrFaust92) [reply aws/resource_aws_codebuild_report_group.go:168]: your right, removing

**Comment by DrFaust92** [comment]
rebased

```
--- PASS: TestAccAWSCodeBuildReportGroup_export_s3 (121.99s)
--- PASS: TestAccAWSCodeBuildReportGroup_tags (89.06s)
--- PASS: TestAccAWSCodeBuildReportGroup_disappears (28.05s)
--- PASS: TestAccAWSCodeBuildReportGroup_basic (37.50s)
```

**Pull request merged**